staging/src/k8s.io/apiserver/pkg/server/storage/resource_config.go

```diff
@@ -26,15 +33,40 @@ type APIResourceConfigSource interface {
 	AnyResourceForGroupEnabled(group string) bool
 }
 
+// GroupVersionRegistry provides access to registered group versions.
+type GroupVersionRegistry interface {
+	// IsGroupRegistered returns true if given group is registered.
+	IsGroupRegistered(group string) bool
+	// IsVersionRegistered returns true if given version is registered.
+	IsVersionRegistered(v schema.GroupVersion) bool
+	// PrioritizedVersionsAllGroups returns all registered group versions.
+	PrioritizedVersionsAllGroups() []schema.GroupVersion
+	// GroupVersionLifecycle returns the APILifecycle for the GroupVersion.
+	GroupVersionLifecycle(gv schema.GroupVersion) schema.APILifecycle
+	// ResourceLifecycle returns the APILifecycle for the GroupVersionResource.
+	ResourceLifecycle(gvr schema.GroupVersionResource) schema.APILifecycle
+}
```

Comment:
nit: curious why not take a scheme

Reply:
Just minimal interface on top of what exists in `staging/src/k8s.io/apiserver/pkg/server/resourceconfig/helpers.go`